### DISCUSSION TITLE
Could `tagModifier` just strip `v` by default?

### DISCUSSION THREAD

**Discussion by ckipp01** [lefou/mill-vcs-version#51] [Ideas]
While I didn't notice it in the beginning, the README has this in it:

> tagModifier: String ⇒ String - allows to modify the git tag, e.g. removing a prefix "v"

It's my fault I didn't catch this, but the fact that this wasn't done by default caught me off guard and caused me to publish both of the Mill plugins I've created so far with `v`s when that's obviously not what I wanted. In the sbt world this is done by default by plugins like https://github.com/sbt/sbt-dynver and I'm assuming that those new to Mill will probably expect this behavior when using a similiar-ish plugin like `mill-vcs-version`. While it's totally doable to just have this in your `build.sc`:

```
  override def publishVersion = VcsVersion
    .vcsState()
    .format(tagModifier = {
      case t if t.startsWith("v") && Try(t.substring(1, 2).toInt).isSuccess =>
        t.substring(1)
      case t => t
    })
```
I find this a bit annoying to include in every project I'm going to be publishing in Mill. Plus, the current behavior seems to break my assumptions which then leads me to wonder if it should just be done by default. wdyt?

**Comment by lefou** [comment]
Personally, I never really understood the reasoning behind the `v` prefix in the git world and found it to cumbersome to use.

But I'm open to this change, if there is a consensus, that it is what the user expects. I could imagine that if a project has the policy, to use a `v` prefix for a version, then it probably doesn't want to treat a non-prefixed tag as version.

**Reply by ckipp01** [reply]
> Personally, I never really understood the reasoning behind the v prefix in the git world and found it to cumbersome to use.

I don't have strong feelings at all about it, it's just every open source project I've personally been apart of doing releases for used it. So I've just been using it by convention I guess.

To elaborate a bit more on why I think it's a pretty safe change is that I can't think of a situation where a user has a `v1.2.1` and they want to publish a literal `v1.2.1` instead of `1.2.1`. As far as I know, no one actually publishes that way. So by stripping the `v` by default, stating that it's the behavior, it then allows for the current normal usecase of not including the `v` to still work fine, but also avoids the boilerplate for projects that use the `v` but then need to strip it.

If we reach consensus on this I'm happy to pr the change.

**Reply by lefou** [reply]
Yeah. As we have no way in the API to block a tag anyways, just stripping the prefix when it is followed by a number sounds useful to me. Looking forward to a pull request.

**Reply by lefou** [reply]
Implemented in version `0.2.0`
